### PR TITLE
Return 404 on HEAD request if source not found

### PR DESCRIPTION
**Pull request by stefanv**
Also require a message to be passed when calling a handler's
`error` method.

### CODE REVIEW

File: skyportal/handlers/base.py [hunk 1]
```diff
@@ -12,5 +12,5 @@ def associated_user_object(self):
     def success(self, *args, **kwargs):
         super().success(*args, **kwargs, extra={'version': __version__})
 
-    def error(self, *args, **kwargs):
-        super().error(*args, **kwargs, extra={'version': __version__})
+    def error(self, message, *args, **kwargs):
```
Comment (acrellin) [inline skyportal/handlers/base.py:15]: Why is this needed?

**Review by acrellin — APPROVED** [review]

File: skyportal/handlers/api/source.py [hunk 2]
```diff
@@ -87,7 +87,8 @@ def head(self, obj_id=None):
         if num_s > 0:
             return self.success()
         else:
-            return self.error()
+            self.set_status(404)
```
Comment (acrellin) [inline skyportal/handlers/api/source.py:90]: Candidate handler probably also needs this

Comment (dmitryduev) [reply skyportal/handlers/api/source.py:90]: Yes, please add the same for candidates.HEAD.

**Review by dmitryduev — CHANGES_REQUESTED** [review]
👍

File: skyportal/handlers/base.py (hunk 1 above)
Comment (stefanv) [reply skyportal/handlers/base.py:15]: It's not strictly speaking needed.

File: skyportal/handlers/api/source.py (hunk 2 above)
Comment (stefanv) [reply skyportal/handlers/api/source.py:90]: Done.

**Review by dmitryduev — APPROVED** [review]
👍
